lib/Sharing/DeckShareProvider.php

```diff
@@ -710,6 +711,13 @@ public function getSharedWith($userId, $shareType, $node, $limit, $offset): arra
 			}
 
 			$qb = $this->dbConnection->getQueryBuilder();
+			// Avoid using implicit cast in order to make use of the index in the join on MySQL/MariaDB
+			// FIXME: Once >= Nextcloud 24 this can be dropped due to https://github.com/nextcloud/server/pull/30471
+			if ($this->dbConnection->getDatabasePlatform() instanceof MySQLPlatform) {
+				$cardIdExpression = $qb->createFunction('CAST(dc.id as CHAR)');
+			} else {
+				$cardIdExpression = $qb->expr()->castColumn('dc.id', IQueryBuilder::PARAM_STR);
+			}
```

Comment:
For newer server releases this will be taken care of in the query builder abstraction with https://github.com/nextcloud/server/pull/30471, so I'd say we stick to those now for the bugfix only releases.